keps/sig-architecture/2527-clarify-status-observations-vs-rbac/README.md

```diff
@@ -0,0 +1,697 @@
+cloud controller sets `Service.spec.control.loadBalancer.ingress.ip` in
+response.  If the user sets `Service.spec.loadBalancerIP` to a specific value,
+the cloud controller will either successfully use that IP and and set it as
+before (ACK), or it will not set any value in
+`Service.spec.control.loadBalancer.ingress` (NAK).
+
+2) Given a Pod, user patches `Pod.spec.containers[0].resources.requests[cpu]`
+to a new value.  Kubelet sees this as a request and, if viable, sets
+`Pod.spec.control.containerStatuses[0].resources.requests[cpu]` to the same
+value.
+
+3) User creates a `Bucket`, setting `Bucket.spec.instance = artifacts-prod`.
+The bucket controller verifies that the namespace is allowed to use the
+artifacts-prod bucket and, if so, sets `Bucket.spec.control.instance` to the same
+value.
+
+#### Tradeoffs
+
+Pro: Retains purity of status.
+
+Con: Confuses the meaning of spec.
+
+Con: Can collide with existing uses of the field name.
+
+Con: Needs a whole new access model.
+
+Con: Likely needs a new subresource.
+
+#### Notes
+
+This model is included for completeness.  I do not expect ANYONE to endorse it.
+
+### Option 4: Use 2 objects
+
+Keep and strengthen the idea that status fields must be from observation.
+Segregate controller-owned fields to a new object, parallel to the user-owned
+object.  RBAC the new object.
+
+#### Examples
+
+1) User creates a Service "foo" and sets `Service.spec.type = LoadBalancer`.
+The cloud controller creates a new ServiceLoadBalancer object "foo" and sets
+`ServiceLoadBalancer.ingress.ip` in response.  If the user sets
+`Service.spec.loadBalancerIP` to a specific value, the cloud controller will
+either successfully use that IP and and set it as before (ACK), or it will
+create a ServiceLoadBalancer object with an error status (NAK).
+
+2) Given a pod "foo", user patches
+`Pod.spec.containers[0].resources.requests[cpu]` to a new value.  Kubelet sees
+this as a request and, if viable, updates the matching PodResources object
+"foo", setting `PodResources.requests[cpu]` to the same value.
+
+3) User creates a `Bucket`, setting `Bucket.spec.instance = artifacts-prod`.
+The bucket controller verifies that the namespace is allowed to use the
+artifacts-prod bucket and, if so, creates a new BucketBinding object which sets
+`BucketBinding.instance` to the same value.
+
+#### Tradeoffs
+
+Pro: RBAC is clear.
+
+Pro: Different controllers can be RBAC'ed to different result-kinds.
+
+Con: Proliferation of many small objects, lifecycle management.
+
+Con: Results are not always 1:1 with user objects (e.g. a Service can have N
+ports, each needs a NodePort allocation).
+
+Con: More watch streams overall (but less traffic on each).
+
+Pro: Feels natural for things like Buckets.
+
+Con: Feels clumsy for things like Pod resources.
+
+Pro: Third-party controllers can follow the same pattern for non-standard
+results.
+
+#### Notes
+
+Even if we ultimately adopt option 1, 2, or 3, option 4 may still be a better
```

Comment:
This option is good for a plugin situation. I don't think it is very related to the other options. I agree with this line but still overall think it is confusing for this option to appear in this document.